From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 14:09:53 -0700
Subject: [PATCH] mm/damon/core: handle damon_call() requests as normal in
 DAMOS watermarks deactivation loop

With out this, DAMOS watermarks-deactivated kdamond cannot be activated
via sysfs commit

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6be802e6e9db..8e0555f65e20 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2543,9 +2543,8 @@ static void kdamond_usleep(unsigned long usecs)
  *
  * If there are &struct damon_call_control requests that registered via
  * &damon_call() on @ctx, do or cancel the invocation of the function depending
- * on @cancel.  @cancel is set when the kdamond is deactivated by DAMOS
- * watermarks, or the kdamond is already out of the main loop and therefore
- * will be terminated.
+ * on @cancel.  @cancel is set when the kdamond is already out of the main loop
+ * and therefore will be terminated.
  */
 static void kdamond_call(struct damon_ctx *ctx, bool cancel)
 {
@@ -2607,7 +2606,7 @@ static int kdamond_wait_activation(struct damon_ctx *ctx)
 		if (ctx->callback.after_wmarks_check &&
 				ctx->callback.after_wmarks_check(ctx))
 			break;
-		kdamond_call(ctx, true);
+		kdamond_call(ctx, false);
 		damos_walk_cancel(ctx);
 	}
 	return -EBUSY;
-- 
2.39.5

